src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java

```diff
@@ -151,13 +148,20 @@ public Object instanceMainPostParseArgs() {
 
         LoggingUtils.setLoggingLevel(VERBOSITY);  // propagate the VERBOSITY level to logging frameworks
 
-        for (final File f : TMP_DIR) {
-            // Intentionally not checking the return values, because it may be that the program does not
-            // need a tmp_dir. If this fails, the problem will be discovered downstream.
-            if (!f.exists()) f.mkdirs();
-            f.setReadable(true, false);
-            f.setWritable(true, false);
-            System.setProperty("java.io.tmpdir", f.getAbsolutePath()); // in loop so that last one takes effect
+        // set the temp directory as a java property, checking for existence and read/write access
+        final Path p = IOUtils.getPath(TMP_DIR);
+        try {
+            p.getFileSystem().provider().checkAccess(p, AccessMode.READ, AccessMode.WRITE);
+            System.setProperty("java.io.tmpdir", IOUtils.getAbsolutePathWithoutFileProtocol(p));
+        } catch (final AccessDeniedException | NoSuchFileException e) {
+            // TODO: it may be that the program does not need a tmp dir
+            // TODO: if it fails, the problem can be discovered downstream
+            // TODO: should log a warning instead?
+            throw new UserException.BadInput(String.format("--%s (%s) should exists and have read/write access",
```

Comment:
Can you use `UserException.BadTmpDir` instead (here and below)?

Reply:
Also, "should exists" should be "should exist"

Reply:
@magicDGS You probably want to rebase before making this change since I just made some changes to BadTmpDir

Reply:
Using `BadTmpDir` as currently implemented is not really pointing out where the error is: if the property was not set, and fails on checking the access it will point to a different tmp directory. I added a new constructor for the exception with a `Path` to provide the one in use.

Done.